src/jinja2/parser.py

```diff
@@ -1025,6 +1028,11 @@ def flush_data() -> None:
                     else:
                         body.append(rv)
                     self.stream.expect("block_end")
+                elif token.type == "comment_begin":
```

Comment:
note: As mentioned in previous note, we don't check for a possible `self.preserve_comments` instance attribute value. Instead we rely on what the lexer yields.